sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -39,8 +39,10 @@ case class AnalyzeTableCommand(
       val table = sparkSession.table(tableIdent.quotedString)
       val cacheManager = sparkSession.sharedState.cacheManager
       if (cacheManager.lookupCachedData(table.logicalPlan).isDefined) {
-        // To collect table stats, materializes an underlying columnar RDD
-        table.collect()
+        if (!noscan) {
+          // To collect table stats, materializes an underlying columnar RDD
+          table.collect()
```

Comment:
According to the pointer you gave in the previous comment, `table.count` is enough for this?

Reply:
oh! I missed it... sorry.